app/src/main/java/com/github/onlynotesswent/model/folder/FolderRepositoryFirestore.kt

```diff
@@ -199,39 +350,120 @@ class FolderRepositoryFirestore(private val db: FirebaseFirestore) : FolderRepos
     }
   }
 
-  override fun deleteFolderContents(
+  override suspend fun deleteFolderContents(
       folder: Folder,
       noteViewModel: NoteViewModel,
       onSuccess: () -> Unit,
-      onFailure: (Exception) -> Unit
+      onFailure: (Exception) -> Unit,
+      useCache: Boolean
   ) {
+    // Update the cache if needed
+    if (useCache) {
```

Comment:
It's probably better to include the cache-induced differences in the same call to `getSubFoldersOf`, to not have a large amount of duplicate calls (will call repository recursively twice)

Reply:
You are right, I'll change this